From 4dda470eacdc726f61d792a6067085d81de6ab50 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Fri, 9 Mar 2018 18:54:49 +0100
Subject: [PATCH 1/1] 51-dracut-rescue.install: fix for initramfs not generated
 in /boot

Commit 5e574046e76e ("5?-dracut*.install: Allow scripts to install
the initramfs in /boot dir") added support to generate initramfs
images in the /boot directory and copy the respective BLS files.

Unfortunately, it broke the rescue initramfs generation when it's
not installed on /boot, due checking for wrong conditions. Fix it.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 51-dracut-rescue.install | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/51-dracut-rescue.install b/51-dracut-rescue.install
index 3f07a698f2f9..679e94b1b8eb 100755
--- a/51-dracut-rescue.install
+++ b/51-dracut-rescue.install
@@ -63,7 +63,7 @@ if ! [[ ${BOOT_OPTIONS[*]} ]]; then
     exit 1
 fi
 
-if [[ -d "$BOOT_DIR_ABS" ]]; then
+if [[ -d "${BOOT_DIR_ABS%/*}" ]]; then
     BOOT_DIR="/${MACHINE_ID}/0-rescue"
     BOOT_ROOT=${BOOT_DIR_ABS%$BOOT_DIR}
     LOADER_ENTRY="$BOOT_ROOT/loader/entries/${MACHINE_ID}-0-rescue.conf"
@@ -106,7 +106,7 @@ case "$COMMAND" in
             ((ret+=$?))
         fi
 
-        if [[ -d "$BOOT_DIR" ]]; then
+        if [[ "${BOOT_DIR_ABS}" != "/boot" ]]; then
             {
                 echo "title      $PRETTY_NAME - Rescue Image"
                 echo "version    $KERNEL_VERSION"
-- 
2.14.3

